### ISSUE TITLE
Rule proposal: No useless template string, or prefer String()

### ISSUE THREAD

**Issue by marcospgp** [eslint/eslint#12866]
**Please describe what the rule should do:**
Disallow useless template strings, such as `${x}`. The thing is this can be used as a way to cast a variable to string, which means the rule could instead be "prefer String()" or "prefer .toString()".

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[X] Suggests an alternate way of doing something (suggestion)
[ ] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
const x = "soomething";
callAFunction(123, "mamamia", `${x}`);
```

```js
const x = 19
callAFunction(123, "mamamia", `${x}`);
```

**Why should this rule be included in ESLint (instead of a plugin)?**

Because eslint already had tons of `no-useless-something` rules, so this one would fit very well with the rest.

**Are you willing to submit a pull request to implement this rule?**
No unless you point me to the right resources and ask nicely and I have enough free time this week

**Comment by mdjermanovic** [comment]
Hi @marcospgp , thanks for the rule proposal!

This might as well be a new option in [no-implicit-coercion](https://eslint.org/docs/rules/no-implicit-coercion)?

**Comment by marcospgp** [comment]
Hey @mdjermanovic , seems like a good fit.

**Comment by anikethsaha** [comment]
Reopening as there is an active PR for this

**Comment by anikethsaha** [comment]
+1 to have it as an option for `no-implicit-coercion`

**Comment by ljharb** [comment]
This should be reopened since there’s an active PR for it.

**Comment by yeonjuan** [comment]
Reopened. There is an active PR #13579

**Comment by remcohaszing** [comment]
This should be reopened. There's still an active PR that's awaiting feedback from the ESLint team.
